File: ios/SagaiSambaandh/Views/PlansView.swift
import SwiftUI

struct PlansView: View {
    @EnvironmentObject var session: SessionManager
    @State private var billingCycle: Int = 0 // 0 = Monthly, 1 = Annual (20% off)
    @State private var alertMessage: String? = nil
    
    private var starterPrice: Int {
        billingCycle == 0 ? 4999 : 3999
    }
    
    private var silverPrice: Int {
        billingCycle == 0 ? 11999 : 9599
    }
    
    private var goldPrice: Int {
        billingCycle == 0 ? 24999 : 19999
    }
    
    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 20) {
                // Header block
                VStack(alignment: .leading, spacing: 6) {
                    Text("Regal Memberships")
                        .font(BrandFonts.displayBold(size: 26))
                        .foregroundColor(.royalMaroon)
                    
                    Text("Select a Rajputana subscription tier to unlock premium features and direct family contact lines.")
                        .font(BrandFonts.body(size: 13))
                        .foregroundColor(.gray)
                }
                .padding(.horizontal, 20)
                .padding(.top, 15)
                
                // Billing Cycle Segment Switcher
                Picker("Billing Cycle", selection: $billingCycle) {
                    Text("Monthly").tag(0)
                    Text("Annual (Save 20%)").tag(1)
                }
                .pickerStyle(SegmentedPickerStyle())
                .padding(.horizontal, 20)
                .padding(.vertical, 5)
                
                // Plans List
                VStack(spacing: 25) {
                    // Plan 1: Starter
                    PlanCard(
                        title: "Starter",
                        price: starterPrice,
                        cycle: billingCycle == 0 ? "month" : "month (billed annually)",
                        features: [
                            "View complete Rajput profiles",
                            "Send 10 express interests / month",
                            "Astrology & Kundli matches overview"
                        ],
                        buttonText: "Select Starter",
                        isFeatured: false
                    ) {
                        subscribe(to: "Starter")
                    }
                    
                    // Plan 2: Rajputana Silver (Featured)
                    PlanCard(
                        title: "Rajputana Silver",
                        price: silverPrice,
                        cycle: billingCycle == 0 ? "month" : "month (billed annually)",
                        features: [
                            "Unlock 15 contact phone numbers",
                            "Astrology compatibility matching reports",
                            "Express interests: Unlimited",
                            "Highlight profile in search lists"
                        ],
                        buttonText: "Choose Silver",
                        isFeatured: true
                    ) {
                        subscribe(to: "Silver")
                    }
                    
                    // Plan 3: Rajputana Gold (Elite)
                    PlanCard(
                        title: "Rajputana Gold",
                        price: goldPrice,
                        cycle: billingCycle == 0 ? "month" : "month (billed annually)",
                        features: [
                            "Direct WhatsApp access to family cards",
                            "Dedicated Rajput matchmaking manager",
                            "Gotra & Kul verification reviews",
                            "Top placement in featured sections"
                        ],
                        buttonText: "Go Gold Elite",
                        isFeatured: false
                    ) {
                        subscribe(to: "Gold")
                    }
                }
                .padding(.horizontal, 20)
                .padding(.bottom, 30)
            }
        }
        .background(Color.sandstoneIvory.edgesIgnoringSafeArea(.all))
        .navigationTitle("Regal Plans")
        .navigationBarTitleDisplayMode(.inline)
        .alert(item: Binding<AlertItem?>(
            get: { alertMessage != nil ? AlertItem(message: alertMessage!) : nil },
            set: { alertMessage = $0?.message }
        )) { alertItem in
            Alert(title: Text("Sagai Sambaandh"), message: Text(alertItem.message), dismissButton: .default(Text("Khammaghani")))
        }
    }
    
    private func subscribe(to tier: String) {
        guard let user = session.currentUser else {
            alertMessage = "Please log in or register your profile to select a membership plan!"
            return
        }
        
        let updatedUser = User(
            id: user.id,
            name: user.name,
            email: user.email,
            gender: user.gender,
            clan: user.clan,
            tier: tier,
            shortlistedIds: user.shortlistedIds,
            unlockedIds: user.unlockedIds
        )
        session.login(user: updatedUser)
        alertMessage = "Congratulations! You have successfully upgraded to the Rajputana \(tier) Membership plan."
    }
}

struct AlertItem: Identifiable {
    var id: String { message }
    let message: String
}

struct PlanCard: View {
    let title: String
    let price: Int
    let cycle: String
    let features: [String]
    let buttonText: String
    let isFeatured: Bool
    var onSelect: () -> Void
    
    var body: some View {
        VStack(spacing: 20) {
            VStack(spacing: 8) {
                if isFeatured {
                    Text("MOST POPULAR")
                        .font(BrandFonts.label(size: 9))
                        .foregroundColor(.white)
                        .padding(.horizontal, 10)
                        .padding(.vertical, 4)
                        .background(Color.royalGold)
                        .cornerRadius(10)
                        .offset(y: -22)
                        .padding(.bottom, -15)
                }
                
                Text(title)
                    .font(BrandFonts.displayBold(size: 22))
                    .foregroundColor(isFeatured ? .white : .royalMaroon)
                
                HStack(alignment: .bottom, spacing: 2) {
                    Text("₹\(price)")
                        .font(BrandFonts.displayBold(size: 32))
                        .foregroundColor(isFeatured ? .lightGold : .royalMaroon)
                    
                    Text("/\(cycle)")
                        .font(BrandFonts.body(size: 11))
                        .foregroundColor(isFeatured ? .white.opacity(0.8) : .gray)
                        .padding(.bottom, 6)
                }
            }
            .padding(.top, isFeatured ? 10 : 0)
            
            Divider()
                .background(isFeatured ? Color.white.opacity(0.3) : Color.royalGold.opacity(0.3))
            
            VStack(alignment: .leading, spacing: 12) {
                ForEach(features, id: \.self) { feature in
                    HStack(alignment: .top, spacing: 10) {
                        Image(systemName: "checkmark.circle.fill")
                            .foregroundColor(isFeatured ? .lightGold : .royalGold)
                            .font(.system(size: 14))
                            .padding(.top, 1)
                        
                        Text(feature)
                            .font(BrandFonts.body(size: 13))
                            .foregroundColor(isFeatured ? .white : .inkBrown)
                    }
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, 10)
            
            Button(action: onSelect) {
                Text(buttonText)
                    .font(BrandFonts.body(size: 14, weight: .bold))
                    .foregroundColor(isFeatured ? .royalMaroon : .white)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 12)
                    .background(isFeatured ? Color.lightGold : Color.royalMaroon)
                    .cornerRadius(8)
                    .shadow(radius: 2)
            }
        }
        .padding(24)
        .background(isFeatured ? Color.royalMaroon : Color.cardBackground)
        .cornerRadius(16)
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(isFeatured ? Color.royalGold : Color.royalGold.opacity(0.2), lineWidth: 2)
        )
        .shadow(color: Color.black.opacity(isFeatured ? 0.12 : 0.05), radius: 10, x: 0, y: 5)
    }
}

File: android/SagaiSambaandh/app/src/main/java/com/sagaisambaandh/ui/screens/PlansView.kt
package com.sagaisambaandh.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.sagaisambaandh.data.SagaiSessionManager
import com.sagaisambaandh.data.User
import com.sagaisambaandh.ui.theme.*

@Composable
fun PlansView(
    session: SagaiSessionManager,
    modifier: Modifier = Modifier
) {
    var billingCycle by remember { mutableStateOf(0) } // 0 = Monthly, 1 = Annual (20% off)
    var showDialog by remember { mutableStateOf(false) }
    var dialogMessage by remember { mutableStateOf("") }

    val starterPrice = if (billingCycle == 0) 0 else 3999
    val silverPrice = if (billingCycle == 0) 11999 else 9599
    val goldPrice = if (billingCycle == 0) 24999 else 19999

    Column(
        modifier = modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .background(MaterialTheme.colorScheme.background)
            .padding(16.dp)
    ) {
        // Header
        Text(
            text = "Regal Memberships",
            color = LightGold,
            fontSize = 24.sp,
            fontWeight = FontWeight.Bold,
            fontFamily = FontFamily.Serif
        )
        Text(
            text = "Select a Rajputana subscription tier to unlock premium features and direct family contact lines.",
            color = SandstoneIvory.copy(alpha = 0.7f),
            fontSize = 13.sp,
            modifier = Modifier.padding(top = 4.dp, bottom = 16.dp)
        )

        // Billing controller switcher
        TabRow(
            selectedTabIndex = billingCycle,
            containerColor = Color.Transparent,
            contentColor = LightGold,
            divider = {},
            indicator = {},
            modifier = Modifier
                .fillMaxWidth()
                .background(RoyalGold.copy(alpha = 0.1f), shape = RoundedCornerShape(10.dp))
                .padding(4.dp)
        ) {
            Tab(
                selected = billingCycle == 0,
                onClick = { billingCycle = 0 },
                modifier = Modifier
                    .background(
                        if (billingCycle == 0) DeepMaroon else Color.Transparent,
                        shape = RoundedCornerShape(8.dp)
                    )
                    .height(36.dp)
            ) {
                Text(
                    text = "Monthly",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (billingCycle == 0) LightGold else SandstoneIvory.copy(alpha = 0.6f)
                )
            }

            Tab(
                selected = billingCycle == 1,
                onClick = { billingCycle = 1 },
                modifier = Modifier
                    .background(
                        if (billingCycle == 1) DeepMaroon else Color.Transparent,
                        shape = RoundedCornerShape(8.dp)
                    )
                    .height(36.dp)
            ) {
                Text(
                    text = "Annual (Save 20%)",
                    fontSize = 13.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (billingCycle == 1) LightGold else SandstoneIvory.copy(alpha = 0.6f)
                )
            }
        }

        Spacer(modifier = Modifier.height(20.dp))

        // Plans grid
        Column(verticalArrangement = Arrangement.spacedBy(20.dp)) {
            // Plan 1: Starter
            PlanCard(
                title = "Starter",
                price = starterPrice,
                cycle = if (billingCycle == 0) "1 month trial" else "month (billed annually)",
                features = listOf(
                    "View complete Rajput profiles",
                    "Send 10 express interests / month",
                    "Astrology & Kundli matches overview"
                ),
                buttonText = "Select Starter",
                isFeatured = false,
                onSelect = {
                    performSubscribe(session, "Starter") { msg ->
                        dialogMessage = msg
                        showDialog = true
                    }
                }
            )

            // Plan 2: Silver (Featured)
            PlanCard(
                title = "Rajputana Silver",
                price = silverPrice,
                cycle = if (billingCycle == 0) "month" else "month (billed annually)",
                features = listOf(
                    "Unlock 15 contact phone numbers",
                    "Astrology compatibility matching reports",
                    "Express interests: Unlimited",
                    "Highlight profile in search lists"
                ),
                buttonText = "Choose Silver",
                isFeatured = true,
                onSelect = {
                    performSubscribe(session, "Silver") { msg ->
                        dialogMessage = msg
                        showDialog = true
                    }
                }
            )

            // Plan 3: Gold
            PlanCard(
                title = "Rajputana Gold",
                price = goldPrice,
                cycle = if (billingCycle == 0) "month" else "month (billed annually)",
                features = listOf(
                    "Direct WhatsApp access to family cards",
                    "Dedicated Rajput matchmaking manager",
                    "Gotra & Kul verification reviews",
                    "Top placement in featured sections"
                ),
                buttonText = "Go Gold Elite",
                isFeatured = false,
                onSelect = {
                    performSubscribe(session, "Gold") { msg ->
                        dialogMessage = msg
                        showDialog = true
                    }
                }
            )
        }
        Spacer(modifier = Modifier.height(30.dp))

        // Dialog alert
        if (showDialog) {
            AlertDialog(
                onDismissRequest = { showDialog = false },
                title = { Text(text = "Shree Rajput Sagai Sambandh", fontFamily = FontFamily.Serif, fontWeight = FontWeight.Bold) },
                text = { Text(text = dialogMessage) },
                confirmButton = {
                    TextButton(onClick = { showDialog = false }) {
                        Text(text = "Khammaghani", color = RoyalMaroon, fontWeight = FontWeight.Bold)
                    }
                },
                containerColor = CardBackground
            )
        }
    }
}

private fun performSubscribe(
    session: SagaiSessionManager,
    tier: String,
    onResult: (String) -> Unit
) {
    val user = session.currentUser.value
    if (user == null) {
        onResult("Please log in or register your profile to select a membership plan!")
        return
    }

    session.upgradeTier(tier)
    onResult("Congratulations! You have successfully upgraded to the Rajputana $tier Membership plan.")
}

@Composable
fun PlanCard(
    title: String,
    price: Int,
    cycle: String,
    features: List<String>,
    buttonText: String,
    isFeatured: Boolean,
    onSelect: () -> Unit
) {
    val cardBg = if (isFeatured) RoyalMaroon else DeepMaroon
    val textMain = if (isFeatured) Color.White else SandstoneIvory
    val textMutedColor = if (isFeatured) Color.White.copy(alpha = 0.8f) else SandstoneIvory.copy(alpha = 0.7f)

    Card(
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = cardBg),
        border = CardDefaults.outlinedCardBorder().copy(
            brush = Brush.linearGradient(
                colors = if (isFeatured) listOf(RoyalGold, LightGold) else listOf(RoyalGold.copy(alpha = 0.2f), RoyalGold.copy(alpha = 0.4f))
            ),
            width = if (isFeatured) 2.dp else 1.dp
        ),
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(modifier = Modifier.padding(24.dp)) {
            // Header text
            Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
                if (isFeatured) {
                    Text(
                        text = "MOST POPULAR",
                        color = RoyalMaroon,
                        fontWeight = FontWeight.Bold,
                        fontSize = 9.sp,
                        modifier = Modifier
                            .background(LightGold, shape = RoundedCornerShape(8.dp))
                            .padding(horizontal = 8.dp, vertical = 3.dp)
                            .offset(y = (-10).dp)
                    )
                }
            }

            Text(
                text = title,
                color = textMain,
                fontSize = 22.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = FontFamily.Serif,
                textAlign = TextAlign.Center,
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(modifier = Modifier.height(8.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.Center,
                verticalAlignment = Alignment.Bottom
            ) {
                Text(
                    text = if (price == 0) "Free" else "₹$price",
                    color = if (isFeatured) LightGold else LightGold,
                    fontSize = 32.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Serif
                )
                Text(
                    text = "/$cycle",
                    color = textMutedColor,
                    fontSize = 11.sp,
                    modifier = Modifier.padding(bottom = 6.dp, start = 2.dp)
                )
            }

            Spacer(modifier = Modifier.height(15.dp))
            Divider(color = if (isFeatured) Color.White.copy(alpha = 0.2f) else RoyalGold.copy(alpha = 0.2f))
            Spacer(modifier = Modifier.height(15.dp))

            // Features list
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                features.forEach { feature ->
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        verticalAlignment = Alignment.Top
                    ) {
                        Icon(
                            imageVector = Icons.Default.CheckCircle,
                            contentDescription = null,
                            tint = if (isFeatured) LightGold else RoyalGold,
                            modifier = Modifier.size(16.dp)
                        )
                        Spacer(modifier = Modifier.width(10.dp))
                        Text(
                            text = feature,
                            color = textMain,
                            fontSize = 13.sp,
                            lineHeight = 18.sp
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(20.dp))

            // Action button
            Button(
                onClick = onSelect,
                shape = RoundedCornerShape(8.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = if (isFeatured) LightGold else RoyalGold,
                    contentColor = RoyalMaroon
                ),
                modifier = Modifier.fillMaxWidth()
            ) {
                Text(
                    text = buttonText,
                    fontWeight = FontWeight.Bold,
                    fontSize = 14.sp
                )
            }
        }
    }
}
